cmd/ulxly/ulxly.go

```diff
@@ -126,6 +182,294 @@ var DepositsCmd = &cobra.Command{
+				if err.Error() != "not found" {
+					log.Error().Err(err)
+				}
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			if r.Status != 0 {
+				fmt.Printf("Deposit Transaction Successful: %s\n", r.TxHash)
+				break
+			} else if r.Status == 0 {
+				fmt.Printf("Deposit Transaction Failed: %s\n", r.TxHash)
+				fmt.Printf("Try increasing the gas limit:\n")
+				fmt.Printf("Current gas limit: %d\n", gasLimit)
+				fmt.Printf("Cumulative gas used for transaction: %d\n", r.CumulativeGasUsed)
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
+		return nil
+	},
+}
+
+var depositClaimCmd = &cobra.Command{
+	Use:     "deposit-claim",
+	Short:   "Make a uLxLy claim transaction",
+	Long:    "Make a uLxLy claim transaction",
+	Args:    cobra.NoArgs,
+	PreRunE: checkClaimArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := cmd.Context()
+		// Dial the Ethereum RPC server.
+		client, err := ethclient.DialContext(ctx, *ulxlyInputArgs.ClaimRPCURL)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to dial rpc")
+		}
+		defer client.Close()
+
+		bridgeV2, err := ulxly.NewUlxly(common.HexToAddress(*ulxlyInputArgs.ClaimBridgeAddress), client)
+		if err != nil {
+			return err
+		}
+
+		privateKey, err := crypto.HexToECDSA(*ulxlyInputArgs.ClaimPrivateKey)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to retrieve private key")
+		}
+
+		publicKey := privateKey.Public()
+		publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+		if !ok {
+			log.Error().Msg("Error casting public key to ECDSA")
+		}
+
+		fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+
+		// value := big.NewInt(*ulxlyInputArgs.Value)
+		gasLimit := *ulxlyInputArgs.ClaimGasLimit
+		gasPrice, err := client.SuggestGasPrice(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get suggested gas price")
+		}
+
+		toAddress := common.HexToAddress(*ulxlyInputArgs.ClaimAddress)
+
+		chainID, err := client.NetworkID(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get chain ID")
+		}
+
+		signer := types.LatestSignerForChainID(chainID)
+
+		/////////////////////////////////////
+		// This section calls the bridge service RPC URL to get the merkle proofs and exit roots and parses them to the correct formats.
+		bridgeServiceProofEndpoint := fmt.Sprintf("%s/merkle-proof?deposit_cnt=%s&net_id=%s", *ulxlyInputArgs.BridgeServiceRPCURL, *ulxlyInputArgs.ClaimIndex, *ulxlyInputArgs.ClaimNetwork)
+		reqBridgeProof, err := http.Get(bridgeServiceProofEndpoint)
+		if err != nil {
+			log.Error().Err(err)
+		}
+		bodyBridgeProof, err := io.ReadAll(reqBridgeProof.Body) // Response body is []byte
+		if err != nil {
+			log.Error().Err(err)
+		}
+		var bridgeProof BridgeProof
+		err = json.Unmarshal(bodyBridgeProof, &bridgeProof) // Parse []byte to go struct pointer, and shadow err variable
+		if err != nil {
+			log.Error().Err(err).Msg("Can not unmarshal JSON")
+			return err
+		}
+
+		merkleProof := [][32]byte{}
+		rollupMerkleProof := [][32]byte{}
```

Comment:
I've been using `common.Hash` rather than [32]byte{}. I find it more convenience. I think it's worth a try and to standarize around that struct.

Reply:
I agree using `common.Hash` may be more streamlined.. but for this particular interaction with the bridge contract, it seems like the `[32][32]byte{}` type is a requirement. From my understanding, using `common.Hash` required reconverting it back eventually to `[32][32]byte{}` format to comply with the bridge contract's inputs

![image](https://github.com/maticnetwork/polygon-cli/assets/125336262/5d0c3465-68d7-4155-a71e-f96493647095)